### DISCUSSION TITLE
Is it possible to create new empty slide with an image

### DISCUSSION THREAD

**Discussion by StravitoMike** [ShapeCrawler/ShapeCrawler#480] [Q&A] [answered]
Hi, 

I'd like to add a slide with an image, is it possible to e.g. add a new slide last in the presentation and then insert an image ?

**Comment by AdamShakhabov** [comment, marked as answer]
Hello.

You can create a new slide, but adding a picture shape is not available yet.

```c#
using ShapeCrawler;

using var pres = SCPresentation.Open("some.pptx");
var layout = pres.SlideMasters[0].SlideLayouts[0];

var newSlide = pres.Slides.AddEmptySlide(layout);

pres.Save();
```

**Reply by AdamShakhabov** [reply]
@StravitoMike, the new method `IShapeCollection.AddPicture()`  was added. Please, check it when you have time and let me know if doesn't work.

**Reply by mhanon** [reply]
Where can I see more docs/demos on the `AddPicture()` method? Im struggling to get the AddPicture method to work. 

```
  //add image
  pres.Slides.AddEmptySlide(layout);
  var newSlideShape = pres.Slides[1].Shapes;
  using FileStream imageStream = new FileStream(imagePath, FileMode.Open, FileAccess.Read);
  newSlideShape.AddPicture(imageStream);
```

**Reply by Luigicube** [reply]
@mhanon 
What worked for me was to set the picture location (X/Y) and size (Width/Height).

**Reply by AdamShakhabov** [reply]
It also doesn't work for me.

The issue https://github.com/ShapeCrawler/ShapeCrawler/issues/649 was created to fix this.

**Reply by mhanon** [reply]
@Luigicube Would you be able to post a work example?

**Reply by AdamShakhabov** [reply]
@mhanon, do you need an example of how to change the X, Y, Width and Height of the Picture?

**Reply by mhanon** [reply]
@ashahabov Yes if you have a working example that would be great. Im looking at add an image to a slide and set the location and dimensions, but the example I tried above would not work. I'm new to ShapeCrawler.

**Reply by AdamShakhabov** [reply]
```c#
var pres = new Presentation("picture.pptx");
var picture = pres.Slides[0].Shapes.GetByName("Picture 1");

// Increase by 50%
picture.Width = (int)(picture.Width * 1.5);
picture.Height = (int)(picture.Height * 1.5);

// Move 50 pixels to the left
picture.X -= 50;

// Move down 50 pixels
picture.Y += 50;

pres.Save();
```

**Reply by Luigicube** [reply]
@mhanon 
Did you manage to add a picture with Adam's example? My code below if needed.

```
private void AddImage(ISlide slide, string filename, double ratio, int X, int Y)
{
	Image img = Image.FromFile(filename);
	Stream imgStream = ToStream(img, ImageFormat.Png);

	slide.Shapes.AddPicture(imgStream);

	IPicture picture = slide.Shapes.OfType<IPicture>().Last();
	picture.Width = (int)Math.Floor(img.Width * ratio);
	picture.Height = (int)Math.Floor(img.Height * ratio);
	picture.X = X;
	picture.Y = Y;
}
```

**Reply by mhanon** [reply]
@Luigicube @ashahabov Thank you both. I was able to get it working.
